_studio/mfx_lib/shared/src/mfx_mpeg2_enc_common_hw.cpp

```diff
@@ -46,7 +46,7 @@ typedef struct tagENCODE_QUERY_STATUS_DATA_tmp
 
 using namespace MfxHwMpeg2Encode;
 
-mfxStatus MfxHwMpeg2Encode::QueryHwCaps(VideoCORE* pCore, ENCODE_CAPS & hwCaps, mfxU8 codecProfileType)
+mfxStatus MfxHwMpeg2Encode::QueryHwCaps(VideoCORE* pCore, ENCODE_CAPS & hwCaps, mfxVideoParam const * par)
 {
     EncodeHWCaps* pEncodeCaps = QueryCoreInterface<EncodeHWCaps>(pCore);
```

Comment:
beginning of this function is messy...

At least we don't need to use ```else``` after ```return``` and we should also switch to usage of ```MFX_CHECK``` macro. But that should be different commit

```
EncodeHWCaps* pEncodeCaps = QueryCoreInterface<EncodeHWCaps>(pCore);
MFX_CHECK(pEncodeCaps , MFX_ERR_UNDEFINED_BEHAVIOR);

if (pEncodeCaps->GetHWCaps<ENCODE_CAPS>(DXVA2_Intel_Encode_MPEG2, &hwCaps) == MFX_ERR_NONE)
            return MFX_ERR_NONE;

std::unique_ptr<DriverEncoder> ddi(CreatePlatformMpeg2Encoder(pCore));
MFX_CHECK_NULL_PTR1(ddi);
```